include/openenclave/bits/report.h

```diff
@@ -98,30 +98,24 @@ typedef struct _oe_identity
 
     /** The unique ID for the enclave.
       * For SGX enclaves, this is the MRENCLAVE value.
-      * For CyReS TAs, this is a hash of the TA image. */
+      * For CyReS components, this is a hash of the CyReS component binary. */
     uint8_t unique_id[OE_UNIQUE_ID_SIZE];
 
     /** The signer ID for the enclave.
       * For SGX enclaves, this is the MRSIGNER value.
-      * For CyReS TAs, this is a hash of the public key that signed the TA image. */
+      * For CyReS components, this is a hash of the public key that signed the CyReS component image. */
     uint8_t signer_id[OE_SIGNER_ID_SIZE];
 
     /** The Product ID for the enclave.
-     * For SGX enclaves, this is the ISVPRODID value. */
+     * For SGX enclaves, this is the ISVPRODID value. 
+     * For CyReS components, this is a binary encoding of a GUID based on the component name. */
     uint8_t product_id[OE_PRODUCT_ID_SIZE];
 
-    /** The unique ID for the device. 
-     *  For CyReS TAs, this is a U-Boot identifier. */
-    uint8_t device_unique_id[OE_UNIQUE_ID_SIZE];
+    /* For CyReS components, this is the ECC public key issued for the CyReS component by the caller component. */
+    uint8_t public_key[OE_ECC_PUBLIC_KEY_SIZE];
```

Comment:
Does this field contain a ECC public key? If it is, it's probably better to name it accordingly.
Or it could be any other type of key?
Another question is:  is it an option to have an union for covering CyRes specific fields?

Reply:
Currently ECC only, happy to rename. As for the union, we could, but I think it would not give us much memory wise, as there would no be other fields to balance it, as I was using all the other fields for CyRes as well. The only exception being security_version, but that does not seem worth it on its own. Your call.

Reply:
For the naming, I was trying to confirm my understanding of this change.  I am ok if you keep it the same.

I don't feel strong about adding union for it. Thanks for explanation.